builtin/console.mbt

```diff
@@ -26,6 +29,18 @@ pub fn print[T : Show](input : T) -> Unit {
   println(input)
 }
 
+///|
+/// Prints and returns the value of a given expression for quick and dirty debugging.
+/// @alert deprecated "This function is for debugging only and should not be used in production"
+pub fn dump[T](t : T, name? : String, loc~ : SourceLoc = _) -> T {
+  let name = match name {
+    Some(name) => name
+    None => ""
+  }
+  println("dump(\{name}@\{loc}) = \{any_to_string(t)}")
+  t
+}
+
```

Comment:
any_to_string is introduced for arbitrary code transformations where `T` is not sure if it satsifies `Show`. How is this different from `dump[T:Show](...)`
cc @Guest0x0 it makes sense to introduce support of ArgsRepr which is essentialy a stringified repr of args. so that we can do something like this:
```mbt
pub fn dump[T:Show](t : T, name~ : ArgsRepr = _, loc~ : SourceLoc = _) -> T
```

Reply:
@bobzhang One thing that distinguishes Swift's `dump()` and Rust's `dbg!()` is that the former doesn't pose any restrictions on `T` whereas the latter depends on the `Debug` trait (which is exclusively used for log-oriented pretty-printing).

I think the `Show` trait in MoonBit corresponds to the `Display` trait rather than `Debug`, so it makes more sense to add no bounds at all.

OTOH `ArgsRepr` will definitely be a nice addition.